### PR TITLE
Fix namespace issue

### PR DESCRIPTION
**Pull request by dcarrolloreilly**
When running popeye from cmd line and passing the -n option to specify a namespace the lint is run for the entire cluster. It appears the namespace is set in context correctly but as we enter the loop to run thru the scrubbers if we encounter a resource that is not namespaced the context is updated to be cluster-wide with a -. This never gets reset back so the next time through the loop we just keep scanning every namespace. This fix grabs the namespace set, stores it in a var and resets it within the loop so that the namespace flag will work correctly limiting the display.

### CODE REVIEW

File: pkg/popeye.go [hunk 1]
```diff
@@ -256,6 +257,8 @@ func (p *Popeye) lint() (int, int, error) {
 			continue
 		}
 
+		ctx = context.WithValue(ctx, internal.KeyNamespace, orig_ns)
```
Comment (derailed) [inline pkg/popeye.go:260]: Context is immutable. This call will create a brand new context every time. 
I think it's best to only alter the original context if we have a clusterwide resource.

**Review by derailed — CHANGES_REQUESTED** [review]
@dcarrolloreilly Great catch! Thank you for this update!!

File: pkg/popeye.go (hunk 1 above)
Comment (dcarrolloreilly) [reply pkg/popeye.go]: Thanks - being new to Go in general I hope I am not heading off in the wrong direction but if I test conditionally within the loop to see if the namespace has been reset in ctx, and only if it has, mutate it using a function like I have included here seems to work. Was this what you had in mind?

**Review by derailed — CHANGES_REQUESTED** [review]
@dcarrolloreilly Thanks for the update. Please see comment below

File: pkg/popeye.go [hunk 2]
```diff
@@ -256,6 +262,11 @@ func (p *Popeye) lint() (int, int, error) {
 			continue
 		}
 
+		// mutate ctx for namespace only if resource is clusterwide
```
[the remark this answers is not in the document]
Comment (derailed) [reply pkg/popeye.go:265]: So I was experimenting with this a bit more and I think there is a indeed a bigger bug :(
The issue here is the context switch timing is wrong. We need to alter/reset the context further down the stack.
I'll take a closer peek...

**Comment by derailed** [comment]
@dcarrolloreilly I took a pass on this and think this issue was resolved on v0.21.5. Please reopen if that's not the case.

**Pull request closed without merging**